src/Compilers/CSharp/Portable/Symbols/FieldSymbol.cs

```diff
@@ -445,6 +449,26 @@ public virtual int TupleElementIndex
         {
             get
             {
+                Debug.Assert(!(this is TupleElementFieldSymbol));
+                if (ContainingType.IsTupleType)
+                {
+                    var i = NamedTypeSymbol.MatchesCanonicalElementName(Name);
+                    if (i > 0 && i <= ContainingType.Arity && i <= NamedTypeSymbol.ValueTupleRestPosition - 1)
+                    {
+                        WellKnownMember wellKnownMember = NamedTypeSymbol.GetTupleTypeMember(ContainingType.Arity, i);
+
+                        RuntimeMembers.MemberDescriptor relativeDescriptor = WellKnownMembers.GetDescriptor(wellKnownMember);
+                        var found = CSharpCompilation.GetRuntimeMember(ImmutableArray.Create<Symbol>(this.OriginalDefinition),
```

Comment:
>CSharpCompilation.GetRuntimeMember(ImmutableArray.Create<Symbol>(this.OriginalDefinition), [](start = 36, length = 90)

I agree, this looks expensive. For each field we will be doing lookup across all fields over and over and over. It feels like the set of default elements should be already figured out, after all we check presence of each of them and fill gaps with error fields. Perhaps that information should be readily available in ```TupleExtraData```, if not already.